packages/automerge-repo/src/DocHandle.ts

```diff
@@ -43,27 +63,30 @@ export class DocHandle<T> //
           states: {
             idle: {
               on: {
+                // If we're creating a new document, we don't need to load anything
                 CREATE: { target: READY },
+                // If we're accessing an existing document, we need to request it from storage
+                // and/or the network
                 FIND: { target: LOADING },
               },
             },
             loading: {
               on: {
+                // LOAD is called by the Repo if the document is found in storage
                 LOAD: { actions: "onLoad", target: READY },
-                UPDATE: { actions: "onUpdate", target: READY },
+                // REQUEST is called by the Repo if the document is not found in storage
                 REQUEST: { target: REQUESTING },
               },
             },
             requesting: {
               on: {
+                // UPDATE is called by the Repo when we receive changes from the network
                 UPDATE: { actions: "onUpdate", target: READY },
               },
-              after: {
-                [TIMEOUT_DELAY]: { actions: "failTimeout", target: ERROR },
-              },
```

Comment:
rather than trying to address timeout here, we do it in `value()`: Now if you're awaiting `docHandle.value()`, and the DocHandle's `timeoutDelay` expires, the promise is rejected.